test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsEmailAddressControllerSpec.scala

```diff
@@ -28,12 +29,30 @@ import play.api.libs.json.Json
 import play.api.test.DefaultAwaitTimeout
 import play.api.test.Helpers.{await, redirectLocation, status}
 import play.twirl.api.HtmlFormat
-import uk.gov.hmrc.plasticpackagingtax.registration.connectors.DownstreamServiceError
+import uk.gov.hmrc.plasticpackagingtax.registration.config.AppConfig
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.{
+  DownstreamServiceError,
+  ServiceError
+}
 import uk.gov.hmrc.plasticpackagingtax.registration.forms.EmailAddress
-import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.PrimaryContactDetails
+import uk.gov.hmrc.plasticpackagingtax.registration.models.emailverification.{
+  EmailStatus,
+  VerificationStatus
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.models.emailverification.{
+  CreateEmailVerificationRequest,
+  EmailStatus,
+  VerificationStatus
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.{
+  MetaData,
+  PrimaryContactDetails
+}
 import uk.gov.hmrc.plasticpackagingtax.registration.views.html.email_address_page
 import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
 
+import scala.concurrent.Future
+
 class ContactDetailsEmailAddressControllerSpec extends ControllerSpec with DefaultAwaitTimeout {
```

Comment:
Can we also cover the cases where we are throwing errors and when exceptions arise. We don't seem to have tests for those.

Reply:
Done